flytekit/remote/init_remote.py

```diff
@@ -0,0 +1,37 @@
+import typing
+import threading
+
+from flytekit.configuration import Config
+from flytekit.remote.remote import FlyteRemote
+from flytekit.tools.translator import Options
+
+REMOTE_ENTRY = None
+REMOTE_ENTRY_LOCK = threading.Lock()
+
+
+def init_remote(
+    config: Config,
+    default_project: typing.Optional[str] = None,
+    default_domain: typing.Optional[str] = None,
+    data_upload_location: str = "flyte://my-s3-bucket/",
+    default_options: typing.Optional[Options] = None,
+    **kwargs,
+):
+    """
+    Initializes the global remote client. This is required before executing any tasks or workflows remotely via `.remote()` method.
+
+    :param default_project: default project to use when fetching or executing flyte entities.
+    :param default_domain: default domain to use when fetching or executing flyte entities.
+    :param data_upload_location: this is where all the default data will be uploaded when providing inputs.
+        The default location - `s3://my-s3-bucket/data` works for sandbox/demo environment. Please override this for non-sandbox cases.
+    :return:
+    """
+    global REMOTE_ENTRY
+    with REMOTE_ENTRY_LOCK:
```

Comment:
If it is initialized already we should raise an exception sayin init should be called only once?

Reply:
I'll restrict it to be called once only but I don't think multiple initializations are a problem here. 
Instead, if we restrict this function to be called once and the user wants to rerun the Jupyter Notebook, the cell with `init_remote()` should be skipped or it will throw an error. Nevertheless, I don't think it is a big problem to skip it either.